cmd/console/deploy/deploy_test.go

```diff
@@ -0,0 +1,216 @@
+				"url": expectedPipelineURL,
+			})
+
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+
+		viper.Set("apibaseurl", baseURL)
+		viper.Set("apitoken", apiToken)
+		viper.Set("project", projectId)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		cmd, buf := prepareCmd(t, environment, revision)
+
+		ctx := factory.WithValue(context.Background(), cmd.OutOrStdout())
+		err := cmd.ExecuteContext(ctx)
+		require.NoError(t, err)
+
+		tableRows := renderer.CleanTableRows(buf.String())
+
+		expectedHeaders := "PROJECT ID | DEPLOY ID | VIEW PIPELINE"
+		expectedRow := fmt.Sprintf("%s | %d | %s", projectId, expectedPipelineId, expectedPipelineURL)
+
+		require.Equal(t, expectedHeaders, tableRows[0])
+		require.Equal(t, expectedRow, tableRows[1])
+
+		require.True(t, gock.IsDone())
+	})
+
+	t.Run("failed deploy", func(t *testing.T) {
+		viper.Reset()
+		defer viper.Reset()
+		defer gock.Off()
+
+		gock.New(baseURL).
+			Post(triggerEndpoint).
+			Reply(400).
+			JSON(map[string]interface{}{})
+
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+
+		viper.Set("apibaseurl", baseURL)
+		viper.Set("apitoken", apiToken)
+		viper.Set("project", projectId)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		cmd, buf := prepareCmd(t, environment, revision)
+
+		ctx := factory.WithValue(context.Background(), cmd.OutOrStdout())
+		err := cmd.ExecuteContext(ctx)
+		require.NoError(t, err)
+
+		base, _ := url.Parse(baseURL)
+		path, _ := url.Parse(triggerEndpoint)
+		require.Contains(
+			t,
+			buf.String(),
+			fmt.Sprintf("POST %s: 400", base.ResolveReference(path)),
+		)
+
+		require.True(t, gock.IsDone())
+	})
+
+	t.Run("missing base url", func(t *testing.T) {
+		viper.Reset()
+		defer viper.Reset()
+
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+
+		viper.Set("apitoken", apiToken)
+		viper.Set("project", projectId)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		cmd, _ := prepareCmd(t, environment, revision)
+
+		ctx := factory.WithValue(context.Background(), cmd.OutOrStdout())
+		err := cmd.ExecuteContext(ctx)
+		require.EqualError(t, err, "API base URL not specified nor configured")
+	})
+
+	t.Run("missing api token", func(t *testing.T) {
+		viper.Reset()
+		defer viper.Reset()
+
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+		viper.Set("apibaseurl", baseURL)
+		viper.Set("project", projectId)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		cmd, _ := prepareCmd(t, environment, revision)
+
+		ctx := factory.WithValue(context.Background(), cmd.OutOrStdout())
+		err := cmd.ExecuteContext(ctx)
+		require.EqualError(t, err, "missing API token - please login")
+	})
+
+	t.Run("missing project flag", func(t *testing.T) {
+		viper.Reset()
+		defer viper.Reset()
+
+		viper.SetConfigFile("/tmp/.miaplatformctl.yaml")
+		viper.Set("apibaseurl", baseURL)
+		viper.Set("apitoken", apiToken)
+		viper.WriteConfigAs("/tmp/.miaplatformctl.yaml")
+
+		cmd, _ := prepareCmd(t, environment, revision)
+
+		// buf := &bytes.Buffer{}
```

Comment:
Why those lines are commented?

Reply:
Just a comment that I forgot to delete. I have done that only in the other PR.